### PR TITLE
Add docs build & update documentation + syntax fixes (#2)

### LINKED ISSUE #222: Update Documentation Infrastructure for Stravalib
In issue #219 , @samryan18 points out that the documentation is out of date. This is likely because the python hosted service doesn't seem to be operating as it used to operate currently. The docs here are already using sphinx which makes it super easy to port to a new platform quickly! We have two options. 

## [Read the Docs](https://readthedocs.org/): 

This platform will automagically connect to a github repo with sphinx setup and allow you to build documentation. it supports versioning, custom environments and branch development of documentation if needed. 

### Potential Drawbacks:
   * maintainers need to setup an account (you can use github to login) 
   *  if docs are failing sometimes you will have to fix things within read the docs (this is rare but i have to do it maybe once or twice a year for our packages). 
   * Advertisements on the docs - see examples below:
[Earthpy docs using standards sphinx theme](https://earthpy.readthedocs.io/en/latest/) 
[Geopandas docs using read the docs (pydata theme)](https://geopandas.readthedocs.io/en/latest/docs.html )
[GEOPANDAS docs running on gh pages - note the lack of versions at the bottom but also no ads: ](https://geopandas.org/index.html)
   * i've run into memory issues with some dev environments using RTD especially if you use conda

What i like to do with this build is create a local github action that tests the documentation build. often you can see the point of error and correct it in github without even having to touch read the docs. every once in a while you need to go into the read the docs admin interface. the new url for docs would be: stravalib.readthedocs.io 

### Benefits
* Doc build maintained by RTD platform
* All versions of your package documentation including latest available to users
* will be super fast to get up and running

##  2. Github Pages Build
In this scenario we'd build the docs via github actions and push them to a new branch here called gh-pages. We'd turn on github pages on the repo and the new docs url will be stravalib.github.io/ . This could work as well but the onus would be on stravalib maintainers to maintain the sphinx website. it would be a bit more infrastructure development here as well to maintain. This solution would also not provide versioned docs so it would always be "latest". Read the docs keeps that history online which is nice. On the flip side in this scenario no one needs a separate set of permissions to work on doc build issues. So in the long term it could be easier if you want more people to be able to help update builds, docs etc.

### Drawbacks
  * A bit more work up front to setup CI infrastructure for the build (but probably worth it)
  * No versioned built docs available to the user online
 
### Benefits
  * No need for an additional login as it all lives here in github / github pages
  * Easier for anyone to update doc builds and such without need special permissions to authenticate with RTD
  * Because the build happens here it's easier to control the envt and avoid envt issues that you can get with RTD
  * no advertisements on your docs

Gosh, now that i typed this all out i am wondering if a github pages build will be easier for long term maintenance because anyone could fix builds without needing permission to use RTD. @hozn do you have any thoughts on this? i'm happy to work on either option.

### PR DESCRIPTION
**Pull request by lwasser**
This closes #222 and i suspect will also potentially address  #212

clean up the model doc reference page so it's easier to follow (removing this as a task and will followup on this once the docs are merged). Then i can add references to the documentation in the docs and view the cleaned up version live and reassess whether this is needed or not. It was definitely useful for the client module but may be less useful here as each "model" element represents a discrete type of element such as gear, etc and as such it's already nicely organized. 
- [x] remove the old api ref structure
- [x] clean up a few more doc strings - i'm using `pyment`  eg: `pyment -o numpydoc -t -f False -w stravalib/client-test-doc.py`   it's really nice. 
- [x] test again

-----
## Demo of the new docs here if you wish for a preview:
http://stravalib.readthedocs.io/
NOTE: Docstrings are now cleaned up
PR here. Because i didn't think this through, i branched off of master so i will wait to open the docstring PR until this docs pr is merged to keep things clean. 
https://github.com/lwasser/stravalib/pull/5/files

### CODE REVIEW

**Comment by lwasser** [comment]
Ok - i plan to fix docstrings in a separate PR but it is started here and i could finish it here.
I didn't accept the invitation to collaborate quickly enough @hozn so i can't setup the gh-pages element but it should be easy to just turn on github pages on the gh-pages branch in the settings if you want to do it. Or just re-invite me to this repo and i will accept more promptly! 

This is what the docs look like now: 
http://www.leahwasser.com/stravalib/index.html 

Once they are live here i'll pull them down from my repo and we can update the URL to the docs in this repo. From there is should be easier to update docs as needed!

I just have the worry that i may have broken something given i can't run tests properly. i didn't make any code changes but did modify docstrings. 
i also updated a lot of url's pointing to the old strava api and where i could added detail to the docstrings. will do more once we have this first iteration working and I see the CI is building as it should. 

CI is setup to only push to gh-pages and update the live docs website when you merge on master. For all other pr's it will just build the docs but not push to the live documentation.

**Comment by lwasser** [comment]
hey there @hozn just checking in on this PR. I have thought a bit more about it and i may back off the docs on github pages in favor of read the docs because with RTD you have versioned docs for each release. It's an easy fix to this pr - i'd just remove the deployment to gh pages via actions. Then I can setup RTD if you are open to it and add you as an admin there. the build is simple enough that RTD should have no issues building docs.

**Comment by hozn** [comment]
Sorry @lwasser , I've been behind with new job.  I'll review this now!  I do generally like the idea of RTD to keep the docs versioned .... if it's not a hard change ...

**Comment by lwasser** [comment]
** removed the build and push to gh-pages so now docs just build via CI here and will push to RTD once we move RTD to direct to this repo (vs my fork)

File: .github/workflows/build-docs.yml
Comment (yihong0618) [inline .github/workflows/build-docs.yml:6]: only `master` or main maybe better?

File: docs/conf.py
Comment (yihong0618) [inline docs/conf.py:17]: path maybe windows?
you can use `os.sep` or something else

File: setup.py
Comment (yihong0618) [inline setup.py:17]: better
```python
with open(news) as f:
    news = f.read()
```

**Review by yihong0618 — APPROVED** [review]

**Comment by hozn** [comment]
Thank you for setting this up!  I've logged in to RTD so I've got a "hozn" account there now.

**Comment by lwasser** [comment]
wonderful. @hozn i just added you to RTD  as a maintainer. i can switch the repo (it's directed at mine currently)  as soon  as this PR  is merged. Also if you want any cleanup on this pr just say the word. i will do a docstring cleanup  in a separate PR so it's easier for you to look at. it will be a lot of changes and i realize this pr is big.

**Comment by hozn** [comment]
@lwasser , this looks great IMO.  I think you should have authority to merge, so please do so as soon as you feel comfortable.  In general, I love that you're taking an active role here and would like to minimize any paperwork that could get in your way!

**Comment by lwasser** [comment]
@hozn wonderful. I appreciate that. You had added me to the repo with permissions however I  believe i didn't accept it in time. My apologies! Are you able to add me again by chance and I will accept the invitation more promptly. Then I can merge. I will then update the link to the documentation in the repo as well and check in on issues that this resolves! Many thanks.

**Comment by hozn** [comment]
@lwasser , no prob -- you should have another invite!

File: .github/workflows/build-docs.yml
Comment (lwasser) [reply .github/workflows/build-docs.yml:6]: hi @yihong0618 thanks! this is setup to just build the docs on all pr's. this is ideal because if people make changes say to docstrings or if we end up adding examples to docstrings and use doctest to test them it will test the build before we merge and we can see if it passes. it does not update the docs in realtime it just builds them in ci. hope that makes sense.

Comment (lwasser) [reply .github/workflows/build-docs.yml:6]: so in short it will test the docs for every PR. but it won't push them live. read the docs will update the docs when we merge to master/main

File: docs/conf.py
Comment (lwasser) [reply docs/conf.py:17]: @yihong0618 can you please clarify what you are looking for / suggesting here? thank you. this should run across operating systems.

File: setup.py
Comment (lwasser) [reply setup.py:17]: sure. what if we merged this as is and create an issue to update syntax for setup.py and conf.py in a separate issue? this pr is only for updating the documentation to ensure it's current. we can totally make these other changes but because it's a big PR and these changes aren't in scope for these issues - we could make them in a separate pr. In face - @yihong0618 would you like to open an issue about these changes and then submit a PR?

**Comment by lwasser** [comment]
ok merging this!! thank you @hozn @yihong0618 i opened a new issue to address the two items you caught in this pr. Feel free to submit a separate PR for those! This one will close the issue related to setting up - updating documentation.

**Comment by lwasser** [comment]
hey @hozn will you please update the link to the docs to: https://stravalib.readthedocs.io ? I will also redirect the doc build to this repo now that this pr is merged!! 🎆 
Screen Shot 2021-12-09 at 1 11 10 PM

**Comment by hozn** [comment]
@lwasser , done!  sorry for delay!

**Pull request merged**
